frontends/web/src/routes/account/send/send.tsx

```diff
@@ -16,33 +16,93 @@
-import { apiWebsocket } from '../../../utils/websocket';
-import { debug } from '../../../utils/env';
+import reject from '../../../assets/icons/cancel.svg';
+import approve from '../../../assets/icons/checked.svg';
+import { alertUser } from '../../../components/alert/Alert';
+import { Balance, BalanceInterface } from '../../../components/balance/balance';
 import { Button, ButtonLink, Checkbox, Input } from '../../../components/forms';
-import { Guide } from '../../../components/guide/guide';
 import { Entry } from '../../../components/guide/entry';
-import { store as fiat } from '../../../components/rates/rates';
-import { alertUser } from '../../../components/alert/Alert';
+import { Guide } from '../../../components/guide/guide';
 import { Header } from '../../../components/layout';
+import { store as fiat } from '../../../components/rates/rates';
 import Status from '../../../components/status/status';
 import WaitDialog from '../../../components/wait-dialog/wait-dialog';
-import { Balance } from '../../../components/balance/balance';
+import { translate, TranslateProps } from '../../../decorators/translate';
+import { debug } from '../../../utils/env';
+import { apiGet, apiPost } from '../../../utils/request';
+import { apiWebsocket } from '../../../utils/websocket';
+import { isBitcoinBased } from '../utils';
 import FeeTargets from './feetargets';
-import UTXOs from './utxos';
-import approve from '../../../assets/icons/checked.svg';
-import reject from '../../../assets/icons/cancel.svg';
 import * as style from './send.css';
+import { SelectedUTXOProps, UTXOs } from './utxos';
+
+interface SendProps {
+    accounts: AccountProps[];
+    code?: string;
+    deviceIDs: string[];
+}
+
+interface AccountProps {
+    code: string;
+    coinCode: string;
+}
+
+interface ProposedAmountProps {
+    amount: string | null;
+    unit: string | null;
+    conversions: ConversionsProps | null;
+}
+
+interface ConversionsProps {
+    [key: string]: string | null;
+}
+
+interface SignProgressProps {
+    steps: number;
+    step: number;
+}
+
+type Props = SendProps & TranslateProps;
+
+interface State {
+    account: AccountProps | null | undefined;
+    balance: BalanceInterface | null;
+    proposedFee: ProposedAmountProps | null;
+    proposedTotal: ProposedAmountProps | null;
+    recipientAddress: string | null;
+    proposedAmount: ProposedAmountProps | null;
+    valid: boolean;
+    amount: string | null;
+    data: string | null;
+    fiatAmount: string | null;
+    fiatUnit: string;
+    sendAll: boolean;
+    feeTarget: string | null;
+    isConfirming: boolean;
+    isSent: boolean;
+    isAborted: boolean;
+    addressError: string | null;
+    amountError: string | null;
+    dataError: string | null;
+    paired: boolean | null;
+    noMobileChannelError: boolean;
+    signProgress: SignProgressProps | null;
+    signConfirm: boolean | null;
+    coinControl: boolean;
+    activeCoinControl: boolean;
+}
```

Comment:
I did not study (yet) whether the right fields are optional but as above I would prefer to make them really optional (with the `?`) so that you don't have to list them all if you declare the object.